qa/rpc-tests/wallet_listreceived.py

```diff
@@ -435,9 +442,101 @@ def test_received_sapling(self, height):
+            {'address': uao, 'amount': 1, 'memo': my_memo},
+            {'address': uaso, 'amount': 2},
+        ])
+        txid0 = wait_and_assert_operationid_status(self.nodes[1], opid)
+        self.sync_all()
+
+        # Decrypted transaction details should be correct, even though
+        # the transaction is still just in the mempool
+        pt = self.nodes[1].z_viewtransaction(txid0)
+
+        assert_equal(pt['txid'], txid0)
+        assert_equal(len(pt['spends']), 0)
+        assert_equal(len(pt['outputs']), 2)
+
+        # Outputs are not returned in a defined order but the amounts are deterministic
+        outputs = sorted(pt['outputs'], key=lambda x: x['valueZat'])
+        assert_equal(outputs[0]['type'], 'orchard')
+        assert_equal(outputs[0]['address'], uao)
+        assert_equal(outputs[0]['value'], Decimal('1'))
+        assert_equal(outputs[0]['valueZat'], 100000000)
+        assert_equal(outputs[0]['output'], 0)
+        assert_equal(outputs[0]['outgoing'], False)
+        assert_equal(outputs[0]['memo'], my_memo)
+        assert_equal(outputs[0]['memoStr'], my_memo_str)
+
+        assert_equal(outputs[1]['type'], 'orchard')
+        assert_equal(outputs[1]['address'], uaso)
+        assert_equal(outputs[1]['value'], Decimal('2'))
+        assert_equal(outputs[1]['valueZat'], 200000000)
+        assert_equal(outputs[1]['output'], 1)
+        assert_equal(outputs[1]['outgoing'], False)
+        assert_equal(outputs[1]['memo'], no_memo)
+        assert 'memoStr' not in outputs[1]
+
+        self.generate_and_sync(height+3)
+
+        opid = self.nodes[1].z_sendmany(uao, [
+            {'address': uaso, 'amount': Decimal('0.4')}
+        ])
+        txid1 = wait_and_assert_operationid_status(self.nodes[1], opid)
+        self.sync_all()
+
+        pt = self.nodes[1].z_viewtransaction(txid1)
+
+        assert_equal(pt['txid'], txid1)
+        assert_equal(len(pt['spends']), 1) # one spend we can see
+        assert_equal(len(pt['outputs']), 2) # one output + one change output we can see
+
+        spends = pt['spends']
+        assert_equal(spends[0]['type'], 'orchard')
+        assert_equal(spends[0]['spend'], 0)
+        assert_equal(spends[0]['txidPrev'], txid0)
+        assert_equal(spends[0]['outputPrev'], 0)
+        assert_equal(spends[0]['address'], uao)
+        assert_equal(spends[0]['value'], Decimal('1.0'))
+        assert_equal(spends[0]['valueZat'], 100000000)
+
+        outputs = sorted(pt['outputs'], key=lambda x: x['valueZat'])
+        assert_equal(outputs[0]['type'], 'orchard')
+        assert_equal(outputs[0]['address'], uaso)
+        assert_equal(outputs[0]['value'], Decimal('0.4'))
+        assert_equal(outputs[0]['valueZat'], 40000000)
+        assert_equal(outputs[0]['outgoing'], False)
+        assert_equal(outputs[0]['memo'], no_memo)
+
+        assert_equal(outputs[1]['type'], 'orchard')
+        # TODO: scrub change addresses
+        #assert_equal(outputs[1]['address'], '<change>')
```

Comment:
Oh hmm, I guess that's another approach: just don't show an address. If we do this, I'd be inclined to have a `change: true` field and just omit the `address` field entirely (so callers don't have to special-case the field to parse the special string `<change>`).